index.js

```diff
@@ -3232,6 +3232,55 @@
   S.range =
   def('range', {}, [$.Integer, $.Integer, $.Array($.Integer)], range);
 
+  //# groupBy :: (a -> a -> Boolean) -> Array a -> Array (Array a)
+  //.
+  //. Splits its list argument into a list of lists of equal,
+  //. adjacent elements. The equality of the elements is tested with
+  //. a curried function provided by the user in the first argument.
+  //. The behavior of groupBy aims to be equal to Haskell's groupby,
+  //. rather than Ramda's groupBy.
+  //.
+  //. The behavior of groupBy can be surprising for functions that
+  //. aren't reflexive, transitive and symmetric
+  //. (see [equivalence][] relation).
+  //.
+  //. ```javascript
+  //. > S.groupBy(S.equals, [1, 1, 2, 3, 3, 4])
+  //. [[1, 1], [2], [3, 3], [4]]
+  //.
+  //. > S.groupBy(S.equals)('aabccd'.split(''))
+  //. [['a', 'a'], ['b'], ['c', 'c'], ['d']]
+  //.
+  //. > S.groupBy(S.equals)([1, 1, 2, 1, 1])
+  //. [[1, 1], [2], [1, 1]]
+  //.
+  //. > var P3 = S.curry2(function(a, b) { return a * b % 3 === 0; });
```

Comment:
We're allowed to use arrow functions in doctests. This could be `a => b => a * b % 3 === 0`.